notify.plugin.zsh

```diff
@@ -26,41 +26,78 @@ zstyle ':notify:*' error-icon ''
 zstyle ':notify:*' disable-urgent no
 zstyle ':notify:*' activate-terminal no
 zstyle ':notify:*' always-check-active-window no
+zstyle ':notify:*' blacklist-regex ''
+zstyle ':notify:*' enable-on-ssh no
 
 unset plugin_dir
 
+function _zsh-notify-is-command-blacklisted() {
+    local blacklist_regex
+    zstyle -s ':notify:*' blacklist-regex blacklist_regex
+    if [[ -z "$blacklist_regex" ]]; then
+        return 1
+    fi
+    print -rn -- "$last_command" | grep -q -E "$blacklist_regex"
```

Comment:
out of curiosity, how do you use the blacklist for?

Reply:
I use it to disable notifications for interactive programs, which I almost never want to get.
One example is having a terminal with vim open somewhere and killing it manually from another one.
Another example is if the background detection of the plugin fails for some reason, I have an extra safeguard.